multi-node/aws/pkg/cluster/cluster.go

```diff
@@ -42,38 +45,132 @@ func New(cfg *config.Cluster, awsDebug bool) *Cluster {
+	describeStacksInput := cloudformation.DescribeStacksInput{
+		StackName: aws.String(c.ClusterName),
+	}
+
+	//Find out if stack exists already. This determines whether we should do subnet conflict validatio
+	var stackExists bool
+	stackNotExistExpr := regexp.MustCompile(fmt.Sprintf("^ValidationError: Stack with id %s does not exist", c.ClusterName))
+
+	describeStacksOutput, err := cfSvc.DescribeStacks(&describeStacksInput)
+	if err != nil {
+		if stackNotExistExpr.Match([]byte(err.Error())) {
+			//No results for a list operation is not an error!!! (unless your AWS)
+			stackExists = false
+		} else {
+			return "", fmt.Errorf("error describing stack: %v", err)
+		}
+	} else {
+		stackExists = true
+		if len(describeStacksOutput.Stacks) > 1 {
+			return "", fmt.Errorf("expected error: found more the one stack with name %s", c.ClusterName)
+		}
+	}
+
+	//if the stack already exists, is doesn't make sense to validate the existing VPC for subnet conflicts
+	if c.VPCID != "" && !stackExists {
+		fmt.Println("Existing VPC detected. Will validate for subnet cidr conflicts")
+		if err := c.validateExistingVPC(); err != nil {
+			return "", err
+		}
+	}
+
+	return validationReport.String(), nil
+}
+
+func (c *Cluster) validateExistingVPC() error {
+	ec2Svc := ec2.New(c.session)
+
+	describeVpcsInput := ec2.DescribeVpcsInput{
+		VpcIds: []*string{aws.String(c.VPCID)},
+	}
+	vpcOutput, err := ec2Svc.DescribeVpcs(&describeVpcsInput)
+	if err != nil {
+		return fmt.Errorf("error describing existing vpc: %v", err)
+	}
+	if len(vpcOutput.Vpcs) == 0 {
+		return fmt.Errorf("could not find vpc %s in region %s", c.VPCID, c.Region)
+	}
+	if len(vpcOutput.Vpcs) > 1 {
+		//Theoretically this should never happen. If it does, we probably want to know.
+		return fmt.Errorf("found more than one vpc with id %s. this is NOT NORMAL.", c.VPCID)
+	}
+
+	existingVPC := vpcOutput.Vpcs[0]
+
+	if *existingVPC.CidrBlock != c.VPCCIDR {
+		//If this is the case, our network config validation cannot be trusted and we must abort
+		return fmt.Errorf("configured vpcCidr (%s) does not match actual existing vpc cidr (%s)", c.VPCCIDR, *existingVPC.CidrBlock)
+	}
+
+	describeSubnetsInput := ec2.DescribeSubnetsInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("vpc-id"),
+				Values: []*string{existingVPC.VpcId},
+			},
+		},
+	}
+
+	subnetOutput, err := ec2Svc.DescribeSubnets(&describeSubnetsInput)
+	if err != nil {
+		return fmt.Errorf("error describing subnets for vpc: %v", err)
+	}
+
+	//Config validation has already ensured this subnet is contained by the existing VPC CIDR
+	//We need access to the net objects so we can detect conflicts
+	subnetIP, subnetNet, err := net.ParseCIDR(c.InstanceCIDR)
+	if err != nil {
+		return fmt.Errorf("error parsing instances cidr %s : %v", c.InstanceCIDR, err)
+	}
+
+	//Loop through all existing subnets in the VPC and look for conflicting CIDRS
+	for _, existingSubnet := range subnetOutput.Subnets {
```

Comment:
This seems to just check overlap of instanceCIDR + VPC CIDR. What about service / pod CIDRS?